clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
As an alternative, I was also considering the following:
```java
void commit(TopicPartition partition, Long offset, String metadata);
```
That's annoying when you need to commit metadata for multiple partitions, especially if you need to provide a callback. Of the options so far, my preference is for `commitWithMetadata`.